### PR TITLE
Gestió dels diacrítics mitjançant preferències

### PR DESCRIPTION
**Pull request by xavivars**
TODO: 
- [x] Edit all modes to support preferences
- [x] Pre-set pre2017 variable on the appropriate modes

### CODE REVIEW

File: modes.xml [hunk 1]
```diff
@@ -143,11 +146,14 @@
...
       </program>
       <program name="apertium-posttransfer"/>
-      <program name="lt-proc $1">
+      <program name="lt-proc $1 -b">
         <file name="spa-cat_valencia_gva.autogen.bin"/>
       </program>
-      <program name="lt-proc -x">
-        <file name="spa-cat.autopgen-diacritics-vells.bin"/>
+      <program name="cg-proc" debug-suff="pre2017prefs">
+        <file name="spa-cat.pre2017_prefs.rlx.bin"/>
```
Comment (unhammer) [inline modes.xml:153]: I guess this is a pipeline with hardcoded prefs (like my nob-nno_e)?

**Review by unhammer — APPROVED** [review]
LGTM, and it works :-)

```sh
$ echo adiós |apertium -d . spa-cat
adeu
$ echo adiós |apertium -d . spa-cat_pre2017
adéu
$ echo adiós |AP_SETVAR='pre2017' apertium -d . spa-cat
adéu
```

File: modes.xml (hunk 1 above)
Comment (xavivars) [reply modes.xml:153]: Yes, this is one forcing the old ortography (it's easier having a "mode" for that than always passing the variable everywhere)

File: Makefile.am [hunk 2]
```diff
@@ -236,6 +250,28 @@ $(PREFIX2).t1x.bin: $(BASENAME).$(PREFIX2).t1x
...
+$(MKDIR_P) prefs
+echo '<preferences>' >$@.tmp
+for f in $^; do xmllint --xpath '/preferences/preference' "$$f"; done >>$@.tmp
+echo '</preferences>' >>$@.tmp
+mv $@.tmp $@
+endef
+
+# uncomment when we have bilingual preferences
+# prefs/$(PREFIX1).xml: $(LANG2).preferences.xml $(PREFIX1).preferences.xml
+prefs/$(PREFIX1).xml: prefs/$(LANG2).preferences.xml
```
Comment (xavivars) [inline Makefile.am:271]: Builds the Preferences.xml needed for Apy

**Pull request merged**